translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -9329,31 +9329,67 @@ Instruction *SPIRVToLLVM::transOCLBuiltinFromExtInst(SPIRVExtInst *BC,
 
 // =============================================================================
 // Translate GLSL.std.450 extended instruction
-Value *SPIRVToLLVM::transGLSLExtInst(SPIRVExtInst *ExtInst,
-                                           BasicBlock *BB) {
+Value *SPIRVToLLVM::transGLSLExtInst(SPIRVExtInst *ExtInst, BasicBlock *BB) {
   auto BArgs = ExtInst->getArguments();
   auto Args = transValue(ExtInst->getValues(BArgs), BB->getParent(), BB);
   switch (static_cast<GLSLExtOpKind>(ExtInst->getExtOp())) {
 
+  case GLSLstd450Round:
+  case GLSLstd450RoundEven:
+    // Round to whole number
+    return getBuilder()->CreateUnaryIntrinsic(Intrinsic::rint, Args[0]);
+
+  case GLSLstd450Trunc:
+    // Trunc to whole number
+    return getBuilder()->CreateUnaryIntrinsic(Intrinsic::trunc, Args[0]);
+
+  case GLSLstd450FAbs:
+    // FP absolute value
+    return getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, Args[0]);
+
+  case GLSLstd450SAbs:
+    // Signed integer absolute value
+    return getBuilder()->CreateSAbs(Args[0]);
+
+  case GLSLstd450FSign:
+    // Get sign of FP value
+    return getBuilder()->CreateFSign(Args[0]);
```

Comment:
please check whether it breaks CTS float_control tests

Reply:
This does not break those tests, but I found several other problems from those tests that I have now fixed.